Sources/FutureMap.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  FutureMap.swift
+//  Deferred
+//
+//  Created by Zachary Waldowski on 4/2/16.
+//  Copyright © 2014-2016 Big Nerd Ranch. Licensed under MIT.
+//
+
+extension FutureType {
+
+    public func map<NewValue>(upon executor: ExecutorType, _ transform: Value -> NewValue) -> Future<NewValue> {
```

Comment:
Undocumented public func - should be able to steal the docs for the queue-based version just below it in the file.